### ISSUE TITLE
yapf dropped support for the "verify" keyword - what kind of PR would you suggest?

### ISSUE THREAD

**Issue by aaronemassey** [jorgenschaefer/elpy#2044]
tldr: Should I make a PR to remove use of `yapf`'s `verify` or to pin installing a specific version of `yapf` that supports it, like `0.40.1`?

Howdy howdy! Thanks for semi-maintaining elpy (I know you don't really have the time now-a-days). `yapf` removed support for the `verify` keyword which is currently used by `yapfutil.py:fix_code()`. I see two ways going forward on how to fix the [resulting unexpected keyword argument error](https://github.com/open-mmlab/mmdetection/issues/10962). Either pin the installed `yapf` to a version that supports the `verify` keyword or remove the usage of the `verify` keyword.

I'm partial to removing support for the `verify` keyword but I don't have the context you do.

**Comment by gopar** [comment]
Hey! 

The way to go about it is to check the version of `yapf` and if its a version that supports it (eg version < `0.40.1`) then passing the verify keyword is fine, otherwise omit it from being passed.

That way we minimize breaking projects that have an older version of `yapf` pinned.

Thank you!

**Comment by aaronemassey** [comment]
Makes sense to me and is better than just pinning it. I'll follow-up with a PR shortly.

**Comment by detrout** [comment]
How's this look? I added the guard as suggested by this issue.

Though I do think the block that generates a parse_version function should probably be shared between blackutil and yapfutil.

Maybe either put it in compat?

```
--- a/elpy/yapfutil.py
+++ b/elpy/yapfutil.py
@@ -10,12 +10,24 @@
 YAPF_NOT_SUPPORTED = sys.version_info < (2, 7) or (
     sys.version_info >= (3, 0) and sys.version_info < (3, 4))
 
+# This came from elpy/blackutil.py it should probably be in a library
+try:
+    from pkg_resources import parse_version
+except ImportError:  # pragma: no cover
+    try:
+        from packaging.version import parse as parse_version
+    except ImportError:  # pragma: no cover
+        raise Fault("Neither `packaging` or `pkg_rsources` could be imported., "
+                    "please reinstall Elpy RPC virtualenv with"
+                    " `M-x elpy-rpc-reinstall-virtualenv`", code=400)
 try:
     if YAPF_NOT_SUPPORTED:
         yapf_api = None
     else:
         from yapf.yapflib import yapf_api
         from yapf.yapflib import file_resources
+        from yapf import __version__
+        yapf_version = parse_version(__version__)
 except ImportError:  # pragma: no cover
     yapf_api = None
 
@@ -28,10 +40,14 @@
         raise Fault('yapf not installed', code=400)
     style_config = file_resources.GetDefaultStyleForDir(directory or os.getcwd())
     try:
+        extra = {}
+        if yapf_version < parse_version("0.40.1"):
+            extra = {"verify": False}
+
         reformatted_source, _ = yapf_api.FormatCode(code,
                                                     filename='<stdin>',
                                                     style_config=style_config,
-                                                    verify=False)
+                                                    **extra)
         return reformatted_source
     except Exception as e:
             raise Fault("Error during formatting: {}".format(e),
```

(edited to fix typo in argument name)

**Comment by yasharhon** [comment]
@detrout I'll move the discussion to the other issue I posted, #2056 .
